app/javascript/components/legacy/nonprofits/donate/InfoStep.tsx

```diff
@@ -0,0 +1,379 @@
+					type: 'setSupporter',
+					supporter: values.supporter,
+					address: values.address,
+					next: stepManagerContext.next,
+				});
+			}} initialValues={{ supporter: props.supporter, address: props.address } as FormikFormValues}>
+				<SupporterFields
+					required={props.required}
+					supporter={props.supporter}
+					hideDedication={props.hideDedication}
+					address={props.address}
+					isRecurring={props.isRecurring}
+					weekly={props.weekly}
+					amount={props.amount}
+					currencySymbol={props.currencySymbol}
+					loadingText={props.loadingText}
+					error={props.error}
+					loading={props.loading} />
+			</Formik>
+			<div>DedicationForm</div>
+		</div>
+	);
+}
+
+interface SupporterFieldsProps {
+  loadingText: string;
+  error: string;
+  loading: boolean;
+  required: RequiredFieldsType;
+  supporter: SupporterType;
+  hideDedication: boolean;
+  address: AddressProps;
+  isRecurring: boolean;
+  weekly: boolean;
+  amount: Money;
+  currencySymbol: string;
+}
+
+interface DedicationLinkProps {
+  hideDedication: boolean;
+}
+
+function SupporterFields(props: SupporterFieldsProps): JSX.Element {
+	const { values, setFieldValue, submitForm } = useFormikContext<FormikFormValues>();
+	const { formatMessage } = useIntl();
+	const emailRequired = formatMessage({ id: 'nonprofits.donate.info.supporter.email_required' });
+	const emailTitle = formatMessage({ id: 'nonprofits.donate.info.supporter.email' }) + `${props.required.email ? `${emailRequired}` : ''}`;
+	const firstNameTitle = formatMessage({ id: 'nonprofits.donate.info.supporter.first_name' });
+	const lastNameTitle = formatMessage({ id: 'nonprofits.donate.info.supporter.last_name' });
+	const phoneTitle = formatMessage({ id: 'nonprofits.donate.info.supporter.phone' });
+	const next = formatMessage({ id: 'nonprofits.donate.amount.next' });
+
+	const [address, setAddress] = useState<AddressProps>(props.address);
+	function setAddressFields(field: string, value: string) {
+		setAddress({ ...address, [field]: value });
+		setFieldValue('address', address);
+	}
+
+	const [supporter, setSupporter] = useState<SupporterType>(props.supporter);
+	function setSupporterFields(field: string, value: string) {
+		setSupporter({ ...supporter, [field]: value });
+		setFieldValue('supporter', supporter);
+	}
+
+	return (
+		<>
+			<RecurringMessage isRecurring={props.isRecurring} weekly={props.weekly} currencySymbol={props.currencySymbol} amount={props.amount} />
+			<div className={'u-marginY--10'}>
+				<input type="hidden" name="profile_id" value={props.supporter?.profileId} />
+				<input type="hidden" name="nonprofit_id" value={props.supporter?.nonprofitId} />
+				<fieldset>
+					<input
+						type="email"
+						name="email"
+						title={emailTitle}
+						required={props.required.email}
+						value={supporter?.email}
+						placeholder={emailTitle}
+						onChange={
+							(e) => {
+								setSupporterFields('email', e.target.value);
+							}
+						} />
```

Comment:
Handling input this way avoids using Formik entirely (or in the near future, react-hook-form) which isn't good because those tools give us things that help us. While it's likely not useful to investigate how Formik works at this point since I hope to have it switched over to react-hook-form today, the same principles apply there.

Reply:
You mean it should be something like this?

```suggestion
					<input
						type="email"
						name="email"
						title={emailTitle}
						required={props.required.email}
						value={values.supporter.email}
						placeholder={emailTitle}
						onChange={
							(e) => {
								setFieldValue('supporter.email', e.target.value); // not sure if this is going to work
							}
						} />
```

Reply:
We're talking about this right now :) Look at using the Field Component in Formik. :smile: